engine/src/main/java/org/terasology/rendering/dag/nodes/ShadowMapNode.java

```diff
@@ -37,21 +37,26 @@
 import org.terasology.rendering.world.RenderableWorld;
 import org.terasology.rendering.world.WorldRenderer;
 import org.terasology.rendering.world.WorldRendererImpl;
-import org.lwjgl.opengl.GL11;
-import static org.lwjgl.opengl.GL11.GL_COLOR_BUFFER_BIT;
-import static org.lwjgl.opengl.GL11.GL_CULL_FACE;
-import static org.lwjgl.opengl.GL11.GL_DEPTH_BUFFER_BIT;
-import static org.lwjgl.opengl.GL11.glClear;
 
+import java.beans.PropertyChangeEvent;
 
 /**
+ * This node class generates a shadow map used by the lighting step to determine what's in sight of
+ * the main light (sun, moon) and what isn't, allowing the display of shadows cast from said light.
+ * TODO: generalize to handle more than one light.
+ *
+ * Instances of this class:
+ * - are enabled and disabled depending on the shadow setting in the rendering config.
+ * - in VR mode regenerate the shadow map only once per frame rather than once per-eye.
+ *
  * Diagram of this node can be viewed from:
  * TODO: move diagram to the wiki when this part of the code is stable
  * - https://docs.google.com/drawings/d/13I0GM9jDFlZv1vNrUPlQuBbaF86RPRNpVfn5q8Wj2lc/edit?usp=sharing
  */
-public class ShadowMapNode extends AbstractNode {
+public class ShadowMapNode extends ConditionDependentNode {
     public static final ResourceUrn SHADOW_MAP = new ResourceUrn("engine:sceneShadowMap");
```

Comment:
Checkstyle is flagging a lot of these public static final ResourceUrn variables, inside nodes, as they could be made private or package private. The things is, these could be potentially used by nodes provided by a module. How to avoid the warning then? Or should I simply ignore it?